libsupport/include/katana/HostAllocator.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdlib>
+#include <cstring>
+
+#include "katana/config.h"
+
+namespace katana {
+
+enum class MemoryPinType { Swappable = 0, Pinned = 1 };
+
+class BaseHostAllocator {
```

Comment:
`HostAllocator` The fact that it's a base class isn't meaningful to someone trying understand this type.